core/store/src/adapter/trie_store.rs

```diff
@@ -167,3 +194,103 @@ pub fn get_key_from_shard_uid_and_hash(shard_uid: ShardUId, hash: &CryptoHash) -
     key[8..].copy_from_slice(hash.as_ref());
     key
 }
+
+#[cfg(test)]
+mod tests {
+    use assert_matches::assert_matches;
+    use near_primitives::errors::StorageError;
+    use near_primitives::hash::CryptoHash;
+    use near_primitives::shard_layout::ShardUId;
+
+    use crate::adapter::trie_store::TrieStoreAdapter;
+    use crate::NodeStorage;
+
+    const ONE: std::num::NonZeroU32 = match std::num::NonZeroU32::new(1) {
+        Some(num) => num,
+        None => panic!(),
+    };
+
+    #[test]
+    fn test_trie_store_adapter() {
+        let (_tmp_dir, opener) = NodeStorage::test_opener();
+        let store = TrieStoreAdapter::new(opener.open().unwrap().get_hot_store());
+        let shard_uids: Vec<ShardUId> =
+            (0..3).map(|i| ShardUId { version: 0, shard_id: i }).collect();
+        let dummy_hash = CryptoHash::default();
+
+        assert_matches!(
+            store.get(shard_uids[0], &dummy_hash),
+            Err(StorageError::MissingTrieValue(_, _))
+        );
+        {
+            let mut store_update = store.store_update();
+            store_update.increment_refcount_by(shard_uids[0], &dummy_hash, &[0], ONE);
+            store_update.increment_refcount_by(shard_uids[1], &dummy_hash, &[1], ONE);
+            store_update.increment_refcount_by(shard_uids[2], &dummy_hash, &[2], ONE);
```

Comment:
Why is the data different every time? 

I guess it doesn't matter too much for this test but if it's not too complicated we can stick to the correct pattern of using State - the data is immutable and the key consists of the shard uid and the hash of the data.

Reply:
In this test we purposefully create 3 different entries in 3 different shards.
`increment_refcount` creates a new entry if it did not exist before.

Reply:
Ah cool, I missed that it's different shard uid every time.